File: solutions/training-java/src/main/java/org/example/Exceptions/Exceptions.java
package org.example.Exceptions;

public class Exceptions {
    public static class IllegalOverDraftException extends Exception{
        public IllegalOverDraftException(){
            super("You have reached your overdraft limit!");
        }
    }


    public static class Bank{
        private static int DISPOLIMIT = 1000;
        private double account_balance;

        public void withdraw(double amount) throws IllegalOverDraftException{
            if(amount > DISPOLIMIT){
                throw new IllegalOverDraftException();
            }else{
                if(amount > account_balance){
                    System.out.println("You cant withdraw that amount");
                }else{
                    account_balance -= amount;
                }

            }
        }

        public void deposit(double amount){
            account_balance += amount;
        }

        public double accountBalanceQuery(){
            return account_balance;
        }
    }

    public static void main(String[] args) {
        Bank mybank = new Bank();

        mybank.deposit(100.0);
        System.out.println(mybank.accountBalanceQuery());

        try {
            mybank.withdraw(1000.1);
        }catch (IllegalOverDraftException exception){
            System.err.println(exception.getMessage());
        }

        System.out.println(mybank.accountBalanceQuery());
    }
}

File: solutions/training-kotlin/src/main/kotlin/Exceptions.kt
import com.sun.nio.sctp.IllegalReceiveException

class Bank {
    var account_balance = 0.0
    val MAX_DEPOSIT = 1000

    fun withdraw(amount: Double) {
        if (amount > MAX_DEPOSIT) {
            throw Exception("You can't withdraw that amount")
        } else if (amount > account_balance) {
            throw Exception("You don't have that money")
        } else {
            account_balance -= amount
        }
    }

    fun deposit(amount: Double) {
        account_balance += amount
    }

    override fun toString(): String {
        return "Bank(account_balance=$account_balance)"
    }
}

fun main() {
    var myaccount = Bank()
    myaccount.deposit(100.0)

    myaccount.toString()

    /*myaccount.withdraw(150.0)

    myaccount.toString()*/

    myaccount.withdraw(1500.0)

    myaccount.toString()
}
